src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActions/SuggestedAction.cs

```diff
@@ -96,80 +96,89 @@ protected Task<ImmutableArray<CodeActionOperation>> GetPreviewOperationsAsync(Ca
 
         public void Invoke(CancellationToken cancellationToken)
         {
-            SourceProvider.UIThreadOperationExecutor.Execute(CodeAction.Title, CodeAction.Message, allowCancellation: true, showProgress: true, action: context =>
-            {
-                // If we want to report progress, we need to create a scope inside the context -- the main context itself doesn't have a way
-                // to report progress. We pass the same allow cancellation and message so otherwise nothing changes.
-                using var scope = context.AddScope(allowCancellation: true, CodeAction.Message);
-                using var combinedCancellationToken = cancellationToken.CombineWith(context.UserCancellationToken);
-                Invoke(new UIThreadOperationContextProgressTracker(scope), combinedCancellationToken.Token);
-            });
+            // Fire and forget.  The called method will set up async operation tracking synchronously on 
+            // this thread.
+            _ = InvokeWithOperationTrackingAsync(cancellationToken);
         }
 
-        public void Invoke(IUIThreadOperationContext context)
+        private Task InvokeWithOperationTrackingAsync(CancellationToken cancellationToken)
         {
-            using var scope = context.AddScope(allowCancellation: true, CodeAction.Message);
-            this.Invoke(new UIThreadOperationContextProgressTracker(scope), context.UserCancellationToken);
+            var token = SourceProvider.OperationListener.BeginAsyncOperation($"{nameof(SuggestedAction)}.{nameof(Invoke)}");
```

Comment:
i could move this into InvokeAsync as well, but i somewhat like the explicit separation (And guarantee that the token is created prior to any async switching).

Reply:
ok.  i will move these BeginAsyncOperation bits into the called async method.